source/MRMesh/MRPolylineTrimWithPlane.cpp

```diff
@@ -5,175 +5,217 @@
                 
 
-        if ( params.otherPart )
+    if ( params.otherPart )
+    {
+        const size_t numEdges = polyline.topology.lastNotLoneEdge().undirected() + 1;
+        UndirectedEdgeBitSet otherPartEdges( numEdges );
+        for ( auto ue : undirectedEdges( polyline.topology ) )
         {
-            const size_t numEdges = polyline.topology.lastNotLoneEdge().undirected() + 1;
-            UndirectedEdgeBitSet otherPartEdges( numEdges );
-            for ( auto ue : undirectedEdges( polyline.topology ) )
-            {
-                if ( !posEdges.test( ue ) )
-                    otherPartEdges.set( ue );
-            }
-
-            vMap.clear();
-            params.otherPart->addPartByMask( polyline, otherPartEdges, &vMap, params.otherOutEmap );
-            if ( params.otherOutVmap )
-                *params.otherOutVmap = vMap;
-            if ( params.fillAfterCut )
-                for ( const auto& segment : cutSegments )
-                    params.otherPart->topology.makeEdge( vMap[segment.first], vMap[segment.second] );
+            if ( !posEdges.test( ue ) )
+                otherPartEdges.set( ue );
         }
-        polyline = res;
+
+        vMap.clear();
+        params.otherPart->addPartByMask( polyline, otherPartEdges, &vMap, params.otherOutEmap );
+        if ( params.otherOutVmap )
+            *params.otherOutVmap = std::move( vMap );
+        if ( params.closeLineAfterCut )
+            for ( const auto& segment : cutSegments )
+                params.otherPart->topology.makeEdge( vMap[segment.first], vMap[segment.second] );
     }
+    polyline = std::move( res );
+}
 
-    std::vector<EdgeSegment> extractSectionsFromPolyline( const Polyline3& polyline, const Plane3f& plane, float eps )
-    {
-        std::vector<EdgeSegment> result;
-        if ( polyline.topology.edgeSize() <= 0 )
-            return {};
+void dividePolylineWithPlane( Polyline3& polyline, const Plane3f& plane, const DividePolylineParameters& params /*= {} */ )
+{
+    trimWithPlane( polyline, plane, params );
+}
 
-        Plane3f planePos( plane.n, plane.d + eps );
-        Plane3f planeNeg( -plane.n, -plane.d + eps );
+std::vector<EdgeSegment> extractSectionsFromPolyline( const Polyline3& polyline, const Plane3f& plane, float eps )
+{
+    std::vector<EdgeSegment> result;
+    if ( polyline.topology.edgeSize() <= 0 )
+        return {};
 
-        struct PointPosition
-        {
-            Vector3f p;
-            float distFromPosPlane = {};
-            float distFromNegPlane = {};
-        };
+    Plane3f planePos( plane.n, plane.d + eps );
+    Plane3f planeNeg( -plane.n, -plane.d + eps );
+
+    struct PointPosition
+    {
+        Vector3f p;
+        float distFromPosPlane = {};
+        float distFromNegPlane = {};
+    };
         
-        for ( auto ue : undirectedEdges( polyline.topology ) )
-        {
-            const EdgeId e( ue );
+    for ( auto ue : undirectedEdges( polyline.topology ) )
```

Comment:
This function already checks all edges of the polyline. So it would be best to compute `UndirectedEdgeBitSet` of edges in positive half-space here, return it from here, and just update them with newly split edges in `subdivideWithPlane`.